pkg/k8shandler/cluster.go

```diff
@@ -51,98 +51,88 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
 
```

Comment:
Good work so far! :tada: `CreateOrUpdateElasticsearchCluster` is still 160 lines long and has a lot of code branches. Could you at least move each conditional-body to a separate named function, e.g. one taking care of the current node in progress, another for the scheduled nodes, finally another for the rolling restart.

I really like the idea of grouping in files, e.g. `clusterrestart`. I suggest we move everything that has to do with rolling restart to `rolling_restart.go` and with redeploys/cert_redeploys to `(cert_)redeploy.go`. WDYT?